src/foam/nanos/ruler/predicate/PropertyIsInstance.js

```diff
@@ -37,14 +46,26 @@ foam.CLASS({
     {
       name: 'f',
       javaCode: `
-
+      Class cls;
+      if ( getCachedClass_() != null ) {
```

Comment:
Couldn't this all just be moved to the javaFactory to make this simpler?

Reply:
Yep, moved, I also have to update cache on the setter of "Of" incase someone changes the class during runtime.